lib/cadet/course/course.ex

```diff
@@ -57,6 +56,30 @@ defmodule Cadet.Course do
     end
   end
 
+  @doc """
+  Create group entity with given name
+  """
+  def create_group(name, leader, mentor) do
+    changeset =
+      %Group{}
+      |> Group.changeset(%{name: name})
+      |> put_assoc(:leader, leader)
+      |> put_assoc(:mentor, mentor)
+
+    Repo.insert(changeset)
+  end
+
+  @doc """
+  Adds given student to given group
+  """
+  def add_student_to_group(group, student) do
+    student
+    |> Repo.preload(:group)
+    |> User.changeset()
+    |> put_assoc(:group, group)
```

Comment:
Similar to above, do not use `put_assoc`, instead use the `<relation name>_id` directly

Reply:
Oh okay. Just saw this comment.